src/main/java/de/teamlapen/vampirism/entity/action/EntityActionHandler.java

```diff
@@ -0,0 +1,291 @@
+        } else if (cooldown > 0) {/* action on cooldown */
+            /* calls updateAction() for {@link ILastingAction} & {@link IInstantAction} as long as the actions cooldown last */
+            cooldown--;
+        } else { /* new action */
+            /* sets a new action if cooldown is over */
+            action = getIntelligentAction();
+            cooldown = action.getCooldown(entity.getLevel());
+            preActivation = action.getPreActivationTime();
+            healthForDisruption = entity.getHealth() - (entity.getMaxHealth() * (float) Balance.ea.DISRUPTION_HEALTH_AMOUNT);
+            if (action instanceof ILastingAction) {
+                duration = ((ILastingAction<T>) action).getDuration(entity.getLevel());
+            }
+        }
+    }
+
+    /**
+     * cancels the action by setting {@link preActivation}, {@link duration} and {@link cooldown} to default
+     */
+    public void cancelActivation() {
+        preActivation = -1;
+        duration = -1;
+        cooldown = 100;
+    }
+
+    public void deactivateAction() {
+        deactivateAction(null);
+    }
+
+    /**
+     * reset the given {@link IEntityAction} or if null, reset the {@link action}
+     * 
+     * @param actionIn
+     */
+    @Nullable
+    public void deactivateAction(IEntityAction actionIn) {
+        IEntityAction action = actionIn != null ? actionIn : this.action;
+        if (action instanceof ILastingAction) {
+            ((ILastingAction<T>) action).deactivate(entity);
+        }
+    }
+
+    public void updateAction() {
+        updateAction(null);
+    }
+
+    /**
+     * updates the given {@link IEntityAction} or if null, activates the {@link action}
+     * 
+     * @param actionIn
+     */
+    @Nullable
+    public void updateAction(IEntityAction actionIn) {
+        IEntityAction action = actionIn != null ? actionIn : this.action;
+        if (action instanceof ILastingAction) {
+            ((ILastingAction<T>) action).onUpdate(entity, duration);
+        }
+    }
+
+    public void updatePreAction() {
+        updatePreAction(null);
+    }
+
+    /**
+     * updates the given {@link IEntityAction} pre activation or if null, updates the {@link action}
+     * 
+     * @param actionIn
+     */
+    @Nullable
+    public void updatePreAction(IEntityAction actionIn) {
+        IEntityAction action = actionIn != null ? actionIn : this.action;
+        if (action instanceof ILastingAction) {
+            ((ILastingAction<T>) action).updatePreAction(entity, preActivation);
+        } else if (action instanceof IInstantAction) {
+            ((IInstantAction<T>) action).updatePreAction(entity, preActivation);
+        }
+    }
+
+    public void activateAction() {
+        activateAction(null);
+    }
+
+    /**
+     * activates the given {@link IEntityAction} or if null, activates the {@link action}
+     * 
+     * @param actionIn
+     */
+    @Nullable
+    public void activateAction(IEntityAction actionIn) {
```

Comment:
Is there any scenario where we would want to activate an action that isn't `this.action`